osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+# Set up django configuration
+import django
+from kobo.client.constants import FINISHED_STATES
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+django.setup()
+from kobo.hub.models import Task  # noqa
+
+from osh.hub.scan.models import RetentionPolicySetting  # noqa
+from osh.hub.scan.models import TaskResultsRemoval  # noqa
+
+logger = logging.getLogger("osh.hub.scripts.osh-retention")
+
+
+def delete_tasks_results(tasks_to_delete):
+    for task, retention_policy_setting in tasks_to_delete:
+        try:
+            logger.info('The following task results will be deleted: %s.', task.id)
+            path_to_delete = Task.get_task_dir(task.id)
+            shutil.rmtree(path_to_delete)
+            logging.info('The following folder was deleted:  %s', path_to_delete)
+        except OSError as e:
+            logger.error("Error deleting tasks folder: %s", e)
+            return
+
+        # If the tasks results are applied, we update the database field, so we don't try to delete them again.
+        TaskResultsRemoval.objects.create(task=task, reason=retention_policy_setting)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that it is a personal task.
+def identify_personal_task(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner
+
+
+def main():
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    # Retrieval of all tasks ids that already have the retention policy applied
+    processed_tasks = TaskResultsRemoval.objects.values_list('task', flat=True)
+    # Retrieval of tasks that has not been applied to retention policy to be applied
+    tasks = Task.objects.filter(state__in=FINISHED_STATES).exclude(id__in=processed_tasks)
+
+    # ---------- Section to take parameters from DB
+    retention_personal_tasks = RetentionPolicySetting.objects.get(category="PersonalScan")
```

Comment:
We should have a try/except block here, so that we log an error if the configuration is missing rather than printing traceback to `stderr`:
```
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com systemd[1]: Starting OpenScanHub retention policy enforcer...
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]: Traceback (most recent call last):
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:   File "/usr/sbin/osh-retention", line 89, in <module>
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:     main()
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:   File "/usr/sbin/osh-retention", line 61, in main
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:     retention_personal_tasks = RetentionPolicySetting.objects.get(category="PersonalScan")
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:   File "/usr/lib/python3.6/site-packages/django/db/models/manager.py", line 85, in manager_method
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:     return getattr(self.get_queryset(), name)(*args, **kwargs)
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:   File "/usr/lib/python3.6/site-packages/django/db/models/query.py", line 437, in get
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]:     self.model._meta.object_name
Sep 07 13:13:59 covscan.lab.eng.brq2.redhat.com osh-retention[3894367]: osh.hub.scan.models.DoesNotExist: RetentionPolicySetting matching query does not exist.
Sep 07 13:14:00 covscan.lab.eng.brq2.redhat.com systemd[1]: osh-retention.service: Main process exited, code=exited, status=1/FAILURE
Sep 07 13:14:00 covscan.lab.eng.brq2.redhat.com systemd[1]: osh-retention.service: Failed with result 'exit-code'.
```

Reply:
Done